build/update.js

```diff
@@ -59,8 +68,6 @@ class Update {
 
     // Go through each item and download/save image
     let done = 0
-    console.log(`Fetching images (${items.length})`)
-
     await new Promise(async (resolve, reject) => {
```

Comment:
I want to review this part, because I think there's a cleaner way than the current pattern

Reply:
What I think might make more sense, is this, but I need to give it a shot. I won't be home for a few hours, but I don't want to let the PR get stale. I'm planning on integrating this into warframe-status and for our weapon/warframe/mod sections, hopefully.
🤞 
I just want to remove as much "resolve, reject" logic as possible

```js
await Promise.all(items.map(async (item) => {
  const imageStub = manifest.find(i => i.uniqueName === item.uniqueName).textureLocation.replace(/\\/g, '/')
  const imageUrl = `http://content.warframe.com/MobileExport${imageStub}`
  const basePath = `${__dirname}/../data/img/`
  const filePath = `${basePath}${item.imageName}`
  const sizeBig = ['Warframes', 'Primary', 'Secondary', 'Melee', 'Relics', 'Sentinels', 'Archwing', 'Skins', 'Pets', 'Arcanes']
  const sizeMedium = ['Resources', 'Misc', 'Fish']
  const image = await request({ url: imageUrl, encoding: null })
  const hash = crypto.createHash('md5').update(image).digest('hex')
  const cached = imageCache.find(c => c.uniqueName === item.uniqueName)

  if (!cached || cached.hash !== hash) {
    this.updateCache(item, hash)

    if (sizeBig.includes(item.category)) {
      await sharp(image).resize(512, 342).ignoreAspectRatio().toFile(filePath)
    } else if (sizeMedium.includes(item.category)) {
      await sharp(image).resize(512, 352).ignoreAspectRatio().toFile(filePath)
    } else {
      await sharp(image).toFile(filePath)
    }
    await minify([filePath], basePath, {
      plugins: [
        minifyJpeg(),
        minifyPng({
          quality: '20-40'
        })
      ]
    })
  }
  done++
  bar.tick({
    image: colors.cyan(item.name),
    updated: !cached || cached.hash !== hash ? colors.yellow('(Updated)') : '',
    check: (bar.curr === items.length - 1) ? colors.green('✓') : colors.yellow('-')
  })
  if (done === items.length) return
}))
```

Reply:
i'm happy with the results